Exclusion of empty rows and getMappingFilePath fix for jars

markusbettig opened the pull request.

markusbettig committed:
Excluded empty rows from import

Filtering rows with row.GetFirstCellNum() != -1 prevents empty rows from being imported.

markusbettig committed:
Made getMappingFilePath return empty string if an exception is thrown

To prevent an import from failing if a mapping file path can't be determined, any exception thrown in getMappingFilePath is logged and an empty string returned.

Background: Running a Java Application from within a jar file, the base directory can't be determined by calling ClassLoader.getSystemResource("") because it returns null.

Nicklas2751 commented:
When it's known that a mapping file can't be loaded because it's a part of an jar why don't you just remove the mapping file entry for the table? Mapping file path is an optional parameter and if you don't fill it, it won't be used.

For the empty row removal have you an simple example which shows the problem? It would be grate if you can build an jUnit test with the example in it and try to push this code combinde with the unit test.

markusbettig committed:
Added test for table with empty rows

Notice how the rows are excluded in ExcelImporter.mappToDataclasses(...) at line 347

Nicklas2751 commented:
I cherry picked the commit 7a67041 and 8e9315c into development branch and will include it in version 1.2. I'll also fix the mapping file problem my idea is to replace all old java.io.File with java.nio Files, Paths so the MappingFile can be a part of an JAR.
